### PR TITLE
PUBDEV-5820 - Added Jetty 9 implementation of IFacade

### PR DESCRIPTION
**Pull request by Pscheidl**
https://0xdata.atlassian.net/browse/PUBDEV-5820

### CODE REVIEW

**Comment by Pscheidl** [comment]
Two little todos, otherwise done.

**Comment by michalkurka** [comment]
Pavel, this is a good start. Thank you. Can we do something about the code duplication?

**Comment by Pscheidl** [comment]
@michalkurka I am not sure a common codebase between jetty 8 and 9 is really what we want. The portion of common code might be rather reduced in future by the needs to modify Jetty 8 & 9 code differently. Some  common logic could be extract for sure.

**Comment by Pscheidl** [comment]
@pkozelka Good point with the dependencies. Yes, but that is solveable, but rather dirty and I'd also prefer not to do that. Our goal should be to get rid of this misuse of servlet standard and restore a proper component-oriented frontend. This will introduce much less work with maintenance and is also upgradeable.

The common code might not become common in future, as there will definitely be changes in Jetty 9. There are API-breaking changes in minor releases in Jetty, could be observed in 8.x already.

File: h2o-jetty-9/src/main/java/water/webserver/jetty9/Jetty9ProxyServerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      throw e;
+    } catch (Exception e) {
+      throw new IOException(e);
+    }
+  }
+
+  private void registerHandlers(HandlerWrapper handlerWrapper, ServletContextHandler context, Credentials credentials, String proxyTo) {
+    // setup authenticating proxy servlet (each request is forwarded with BASIC AUTH)
+    final ServletHolder proxyServlet = new ServletHolder(TransparentProxyServlet.class);
+    proxyServlet.setInitParameter("ProxyTo", proxyTo);
```
Comment (pkozelka) [inline h2o-jetty-9/src/main/java/water/webserver/jetty9/Jetty9ProxyServerAdapter.java]: I believe that `ProxyTo` works only on Jetty 8, but Jetty 9 requires `proxyTo` parameter - that's why I originally did it https://github.com/h2oai/h2o-3/pull/2938/commits/a4d26b5cae69b7ba9e6b15873ea38a65521dcee2

File: h2o-jetty-9/src/main/java/water/webserver/jetty9/Jetty9Helper.java [hunk 2]
```diff
@@ -73,6 +72,7 @@ Server createJettyServer(String ip, int port) {
         final ArrayList<ConnectionFactory> connectionFactories = new ArrayList<>();
         ConnectionFactory connectionFactory = new HttpConnectionFactory();
         final HttpConfiguration httpConfiguration = new HttpConfiguration();
+        httpConfiguration.setSendServerVersion(true);
```
Comment (pkozelka) [inline h2o-jetty-9/src/main/java/water/webserver/jetty9/Jetty9Helper.java]: I think this is good idea - and we can add the same to jetty8 version in this PR, WDYT?

Comment (Pscheidl) [reply h2o-jetty-9/src/main/java/water/webserver/jetty9/Jetty9Helper.java]: We already have this in Jetty 8. This is only done in Jetty 9 in a completely different way, like this ^^^.

File: h2o-assemblies/main/build.gradle [hunk 3]
```diff
@@ -16,7 +16,7 @@ configurations {
 // Dependencies
 dependencies {
     compile project(":h2o-app")
-    runtime project(":h2o-jetty-8")
```
Comment (michalkurka) [inline h2o-assemblies/main/build.gradle]: @Pscheidl I would keep the default as Jetty 8 in this PR and only enable it in CDH 6.

**Comment by Pscheidl** [comment]
@michalkurka We will keep the test. As the only module with Jetty 9 functionality will be CDH 6.0, this test may be preserved now.

**Pull request merged**